Sources/Rendering/Core/AxesActor/index.js

```diff
@@ -107,76 +123,70 @@ function vtkAxesActor(publicAPI, model) {
   publicAPI.update();
   const _debouncedUpdate = macro.debounce(publicAPI.update, 0);
 
-  const { setConfig, setXAxisColor, setYAxisColor, setZAxisColor } = publicAPI;
+  publicAPI.setXAxisColor = (color) =>
+    publicAPI.setXConfig({ ...publicAPI.getXConfig(), color });
 
-  publicAPI.setConfig = (c) => {
-    if (setConfig(c)) {
-      _debouncedUpdate();
-      return true;
-    }
-    return false;
-  };
+  publicAPI.setYAxisColor = (color) =>
+    publicAPI.setYConfig({ ...publicAPI.getYConfig(), color });
 
-  publicAPI.setXAxisColor = (c) => {
-    if (setXAxisColor(c)) {
-      _debouncedUpdate();
-      return true;
-    }
-    return false;
-  };
+  publicAPI.setZAxisColor = (color) =>
+    publicAPI.setZConfig({ ...publicAPI.getZConfig(), color });
 
-  publicAPI.setYAxisColor = (c) => {
-    if (setYAxisColor(c)) {
-      _debouncedUpdate();
-      return true;
-    }
-    return false;
-  };
+  publicAPI.getXAxisColor = () => model.getXConfig().color;
 
-  publicAPI.setZAxisColor = (c) => {
-    if (setZAxisColor(c)) {
-      _debouncedUpdate();
-      return true;
-    }
-    return false;
-  };
+  publicAPI.getYAxisColor = () => model.getYConfig().color;
+
+  publicAPI.getZAxisColor = () => model.getZConfig().color;
+
+  model._onConfigChanged = _debouncedUpdate;
+  model._onXConfigChanged = _debouncedUpdate;
+  model._onYConfigChanged = _debouncedUpdate;
+  model._onZConfigChanged = _debouncedUpdate;
 }
 
 // ----------------------------------------------------------------------------
 // Object factory
 // ----------------------------------------------------------------------------
 
-export const DEFAULT_VALUES = {
-  config: {
-    recenter: true,
-    tipResolution: 60,
-    tipRadius: 0.1,
-    tipLength: 0.2,
-    shaftResolution: 60,
-    shaftRadius: 0.03,
-    invert: false,
-  },
-  xAxisColor: [255, 0, 0],
-  yAxisColor: [255, 255, 0],
-  zAxisColor: [0, 128, 0],
-};
+function defaultValues(initialValues) {
+  return {
+    config: {
+      recenter: true,
+      tipResolution: 60,
+      tipRadius: 0.1,
+      tipLength: 0.2,
+      shaftResolution: 60,
+      shaftRadius: 0.03,
+      invert: false,
+      ...initialValues?.config,
+    },
+    xConfig: {
+      color: [255, 0, 0],
+      invert: false,
+      ...initialValues?.xConfig,
+    },
+    yConfig: {
+      color: [255, 255, 0],
+      invert: false,
+      ...initialValues?.yConfig,
+    },
+    zConfig: {
+      color: [0, 128, 0],
+      invert: false,
+      ...initialValues?.zConfig,
+    },
+  };
+}
 
 // ----------------------------------------------------------------------------
 
 export function extend(publicAPI, model, initialValues = {}) {
-  Object.assign(model, DEFAULT_VALUES, initialValues);
+  Object.assign(model, defaultValues(initialValues));
```

Comment:
this line is not needed, if you already pass `defaultValues()` in the `extend()` function below